src/main/java/org/folio/rest/impl/UserGroupAPI.java

```diff
@@ -40,323 +32,176 @@
-
-        PostgresClientUtil.getInstance(vertxContext, okapiHeaders).get(GROUP_TABLE, Usergroup.class,
-          new String[]{"*"}, cql, true, true,
-            reply -> {
-              try {
-                if(reply.succeeded()){
-                  Usergroups groups = new Usergroups();
-                  @SuppressWarnings("unchecked")
-                  List<Usergroup> groupList = reply.result().getResults();
-                  groups.setUsergroups(groupList);
-                  groups.setTotalRecords(reply.result().getResultInfo().getTotalRecords());
-                  asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetGroupsResponse
-                    .respond200WithApplicationJson(groups)));
-                }
-                else{
-                  log.error(reply.cause().getMessage(), reply.cause());
-                  asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetGroupsResponse
-                    .respond400WithTextPlain(reply.cause().getMessage())));
-                }
-              } catch (Exception e) {
-                log.error(e.getMessage(), e);
-                asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetGroupsResponse
-                  .respond500WithTextPlain(messages.getMessage(
-                    lang, MessageConsts.InternalServerError))));
-              }
-            });
-      }
-      catch(FieldException fe){
-        log.error(fe.getLocalizedMessage(), fe);
-        asyncResultHandler.handle(Future.succeededFuture(GetUsersResponse.respond400WithTextPlain(
-                "CQL Parsing Error for '" + query + "': " + fe.getLocalizedMessage())));
-      }
-      catch (Exception e) {
-        log.error(e.getMessage(), e);
-        asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetGroupsResponse
-          .respond500WithTextPlain(messages.getMessage(lang, MessageConsts.InternalServerError))));
-      }
-    });
+    String lang, Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
 
+    PgUtil.get(GROUP_TABLE, Usergroup.class, Usergroups.class, query, offset, limit, okapiHeaders,
+      vertxContext, GetGroupsResponse.class, asyncResultHandler);
   }
 
   @Validate
   @Override
   public void postGroups(String lang, Usergroup entity,
-          Map<String, String> okapiHeaders,
-          Handler<AsyncResult<Response>> asyncResultHandler,
-          Context vertxContext) {
-
-    vertxContext.runOnContext(v -> {
-      try {
-        System.out.println("sending... postGroups");
-        String id = entity.getId();
-        if(id == null) {
-          id = UUID.randomUUID().toString();
-          entity.setId(id);
+    Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    String id = entity.getId();
+    if (id == null) {
+      id = UUID.randomUUID().toString();
+      entity.setId(id);
+    }
```

Comment:
Generating randomUUID() can be handled by PgUtil.post:
 https://github.com/folio-org/raml-module-builder/blob/v25.0.0/domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java#L505